src/core/mod.rs

```diff
@@ -285,6 +320,51 @@ impl BPF {
         Ok(Table::new(id, self.ptr()))
     }
 
+    /// Load a BPF function and return its file descriptor. Useful in BPF tail-calls.
+    pub fn load_func(&mut self, name: &str, bpf_prog_type: BpfProgType) -> Result<i32, BccError> {
+        let name_cstring = CString::new(name).unwrap();
+        let name_ptr = name_cstring.as_ptr();
+
+        if let Some(fd) = self.functions.get(name) {
+            return Ok(*fd);
+        }
+
+        let log_level = if self.debug.contains(BccDebug::BPF_REGISTER_STATE) {
+            2
+        } else if self.debug.contains(BccDebug::BPF) {
+            1
+        } else {
+            0
+        };
+
+        unsafe {
+            let f_start = bpf_function_start(self.ptr(), name_ptr);
+            if f_start as usize == 0 {
+                return Err(BccError::Loading {
+                    name: name.to_string(),
+                    message: String::from("The specified function could not be found."),
+                });
+            }
+
+            let fd = bcc_func_load(
```

Comment:
looks like this doesn't work on all BCC versions we support. Look at the CI logs.

Reply:
Completely missed this, thanks for raising it